src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedEntryPointSymbol.cs

```diff
@@ -450,13 +429,11 @@ private sealed class ScriptEntryPoint : SynthesizedEntryPointSymbol
             //     var script = new Script();
             //     script.<Initialize>().GetAwaiter().GetResult();
             // }
-            internal override BoundBlock CreateBody()
+            internal override BoundBlock CreateBody(DiagnosticBag diagnostics)
             {
-                // CreateBody should only be called if no errors.
-                Debug.Assert((object)_getAwaiterMethod != null);
-                Debug.Assert((object)_getResultMethod != null);
-
                 var syntax = DummySyntax();
+                var compilation = _containingType.DeclaringCompilation;
+                var binder = compilation.GetBinder((CSharpSyntaxNode)_containingType.DeclaringSyntaxReferences[0].SyntaxTree.GetRoot());
```

Comment:
Unlike Main in regular C# file the synthesized entry script point code is not contained/in scope in any single script file. If there are multiple script files in the compilation (via ```#load```) the code of all of them is merged into a single entry point. Each of these files may have different using directives. The only using directives that should be applied for the synthesized entry point are the global directives. These are usually specified by the script host (e.g. thru command line switch in case of csi).